### PR TITLE
Upgrade TensorRT to v7.1.3.4

### PR DESCRIPTION
**Pull request by stevenlix**
1. Upgrade TensorRT to version 7.1.3.4
2. Update onnx-tensorrt parser to support the new TRT version
3. Remove empty shape constraints in TRT EP
4. Fix an issue in cycle detection due to kernel name changes
5. Remove unused initializers in TRT subgraphs

### CODE REVIEW

**Comment by jywu-msft** [comment]
https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=192560&view=logs&j=72b59a5c-f9fa-5437-adfc-3f855724a45f&t=196a7b30-78bb-593c-2d38-e63ead168fa9&l=11

```
C:\hostedtoolcache\windows\Python\3.7.8\x64\python.exe D:\2\s\tools\ci_build\build.py --config Debug --build_dir D:\2\b --skip_submodule_sync --build_shared_lib --update --cmake_generator "Visual Studio 16 2019" --msvc_toolset 14.16 --build_wheel --enable_onnx_tests --use_tensorrt --tensorrt_home=C:\local\TensorRT-7.1.3.4.cuda-11.0.cudnn8.0\TensorRT-7.1.3.4 --cuda_version=11.0 "--cuda_home=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0" --cudnn_home=C:\local\cudnn-11.0-windows-x64-v8.0.2.39\cuda --cmake_extra_defines CMAKE_SYSTEM_VERSION=10.0.18362.0
2020-08-03 23:06:08,979 Build [ERROR] - No version file found in CUDA install directory. Looked for C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0\version.txt

##[error]The process 'C:\hostedtoolcache\windows\Python\3.7.8\x64\python.exe' failed with exit code 1
Finishing: Generate cmake config
```
@snnn , are some CI machines missing CUDA11?

File: tools/ci_build/github/azure-pipelines/win-gpu-tensorrt-ci-pipeline.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ jobs:
     displayName: 'Generate cmake config'
     inputs:
       scriptPath: '$(Build.SourcesDirectory)\tools\ci_build\build.py'
-      arguments: '--config $(BuildConfig) --build_dir $(Build.BinariesDirectory) --skip_submodule_sync --build_shared_lib --update --cmake_generator "Visual Studio 16 2019" --msvc_toolset 14.16 --build_wheel --enable_onnx_tests --use_tensorrt --tensorrt_home="C:\local\TensorRT-7.0.0.11.cuda-10.2.cudnn7.6\TensorRT-7.0.0.11" --cuda_version=10.2 --cuda_home="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.2" --cudnn_home="C:\local\cudnn-10.2-windows10-x64-v7.6.5.32\cuda" --cmake_extra_defines CMAKE_SYSTEM_VERSION=10.0.18362.0'
+      arguments: '--config $(BuildConfig) --build_dir $(Build.BinariesDirectory) --skip_submodule_sync --build_shared_lib --update --cmake_generator "Visual Studio 16 2019" --msvc_toolset 14.16 --build_wheel --enable_onnx_tests --use_tensorrt --tensorrt_home="C:\local\TensorRT-7.1.3.4.cuda-11.0.cudnn8.0\TensorRT-7.1.3.4" --cuda_version=11.0 --cuda_home="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0" --cudnn_home="C:\local\cudnn-11.0-windows-x64-v8.0.2.39\cuda" --cmake_extra_defines CMAKE_SYSTEM_VERSION=10.0.18362.0'
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/win-gpu-tensorrt-ci-pipeline.yml:45]: if we upgrade to CUDA 11.0 , it should contain the fix for the cuda compiler bug right? so we wouldn't need to use msvc_toolset 14.16 ?

File: tools/ci_build/github/azure-pipelines/win-gpu-tensorrt-ci-pipeline.yml [hunk 2]
```diff
@@ -75,7 +75,7 @@ jobs:
...
-     python $(Build.SourcesDirectory)\tools\ci_build\build.py --config $(BuildConfig) --build_dir $(Build.BinariesDirectory) --skip_submodule_sync --build_shared_lib --test --cmake_generator "Visual Studio 16 2019" --msvc_toolset 14.16 --build_wheel --enable_onnx_tests --use_tensorrt --tensorrt_home="C:\local\TensorRT-7.0.0.11.cuda-10.2.cudnn7.6\TensorRT-7.0.0.11" --cuda_version=10.2 --cuda_home="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.2" --cudnn_home="C:\local\cudnn-10.2-windows10-x64-v7.6.5.32\cuda" --cmake_extra_defines CMAKE_SYSTEM_VERSION=10.0.18362.0
+     python $(Build.SourcesDirectory)\tools\ci_build\build.py --config $(BuildConfig) --build_dir $(Build.BinariesDirectory) --skip_submodule_sync --build_shared_lib --test --cmake_generator "Visual Studio 16 2019" --msvc_toolset 14.16 --build_wheel --enable_onnx_tests --use_tensorrt --tensorrt_home="C:\local\TensorRT-7.1.3.4.cuda-11.0.cudnn8.0\TensorRT-7.1.3.4" --cuda_version=11.0 --cuda_home="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0" --cudnn_home="C:\local\cudnn-11.0-windows-x64-v8.0.2.39\cuda" --cmake_extra_defines CMAKE_SYSTEM_VERSION=10.0.18362.0
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/win-gpu-tensorrt-ci-pipeline.yml:78]: same comment as above for msvc_toolset 14.16

File: onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc [hunk 3]
```diff
@@ -427,10 +379,10 @@ SubGraphCollection_t TensorrtExecutionProvider::GetSupportedList(SubGraphCollect
       if (group.second) {
         nodes_list_output.push_back(group);
       } else {
-        onnxruntime::Model model_build(graph.Name(), true, ModelMetaData(), PathString(),
-                                       IOnnxRuntimeOpSchemaRegistryList(), graph.DomainToVersionMap(),
-                                       std::vector<ONNX_NAMESPACE::FunctionProto>(), *GetLogger());
-        onnxruntime::Graph& graph_build = model_build.MainGraph();
+        std::unique_ptr<onnxruntime::Model> model_build = std::make_unique<onnxruntime::Model>(graph.Name(), true, ModelMetaData(), PathString(),
```
Comment (jywu-msft) [inline onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:382]: we can't use std::make_unique() as we're supposed to be c++11 compliant.
that's why we have onnxruntime::make_unique()

**Comment by snnn** [comment]
> https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=192560&view=logs&j=72b59a5c-f9fa-5437-adfc-3f855724a45f&t=196a7b30-78bb-593c-2d38-e63ead168fa9&l=11
> 
> ```
> C:\hostedtoolcache\windows\Python\3.7.8\x64\python.exe D:\2\s\tools\ci_build\build.py --config Debug --build_dir D:\2\b --skip_submodule_sync --build_shared_lib --update --cmake_generator "Visual Studio 16 2019" --msvc_toolset 14.16 --build_wheel --enable_onnx_tests --use_tensorrt --tensorrt_home=C:\local\TensorRT-7.1.3.4.cuda-11.0.cudnn8.0\TensorRT-7.1.3.4 --cuda_version=11.0 "--cuda_home=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0" --cudnn_home=C:\local\cudnn-11.0-windows-x64-v8.0.2.39\cuda --cmake_extra_defines CMAKE_SYSTEM_VERSION=10.0.18362.0
> 2020-08-03 23:06:08,979 Build [ERROR] - No version file found in CUDA install directory. Looked for C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0\version.txt
> 
> ##[error]The process 'C:\hostedtoolcache\windows\Python\3.7.8\x64\python.exe' failed with exit code 1
> Finishing: Generate cmake config
> ```
> 
> @snnn , are some CI machines missing CUDA11?

No, they're not, The installation dir doesn't have version.txt. I don't know why. I manually installed it again, it still doesn't have it. Do you have a local CUDA 11 installation? Could you find the file?

**Comment by stevenlix** [comment]
I have the version.txt in my local CUDA11 installation directory

**Comment by jywu-msft** [comment]
please merge master to pick up @snnn 's change for the version.txt issue for CUDA11

**Pull request merged**